### PR TITLE
Added rule to detect LKM module injects using insmod used by rootkits for kernel hooking

### PR DESCRIPTION
**Pull request by d1vious**
**What type of PR is this?**

/kind rule-create

**Any specific area of the project related to this PR?**

/area rules

**What this PR does / why we need it**:

This PR adds a new falco rule that looks for when insmod is called as part of a execve event. Injecting LKM modules on (post build) running production instances should be rare and is a common way for rootkits that employ kernel hooking to obfuscate themselves. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This is a rehashed of [PR#1389](https://github.com/falcosecurity/falco/pull/1389) just cleaned up DCO

**Does this PR introduce a user-facing change?**:

Yes new rule in falco_rules.yml

```release-note
rule(Linux Kernel Module injection detected): adds a new rule that detects when a LKM module is inject using insmod typically used by rookits looking to obfuscate their behavior via kernel hooking. 
```

### CODE REVIEW

**Comment by leogr** [comment]
Thanks @d1vious !

PS: I have just edited the PR description to fix the release note block syntax.

**Comment by d1vious** [comment]
Thank you 😊

**Review by leodido — COMMENTED** [review]
Thanks @d1vious!

Waiting for @Kaizhe review but just a comment: shouldn't this new macro exclude the Falco kernel module ?

**Comment by d1vious** [comment]
@leodido we can add exclusion for a bunch of commonly used modules, if you like we can collage a list and tack it to the end of the query. Otherwise I suspect a customer would have to do some tuning.

**Review by leogr — COMMENTED** [review]
Thank you for this PR :+1: 

The rule seems good to me, but I believe it worth adding an exclusion list for known modules.
IMHO, the list should at least include the Falco kernel module, then users can extend it.

File: rules/falco_rules.yaml
Comment (Kaizhe) [inline rules/falco_rules.yaml:3014]: Can you please capitalized the beginning letter of every word?

File: rules/falco_rules.yaml
Comment (Kaizhe) [inline rules/falco_rules.yaml:3015]: Can you rephrase a little bit of the description to something as simple as 
```
Detect kernel module was injected (from container).
```

File: rules/falco_rules.yaml
Comment (Kaizhe) [inline rules/falco_rules.yaml:3016]: I would suggest to limit to activities happen from inside a container. So the condition would be something like:
`spawned_process and container and proc.name=insmod`

File: rules/falco_rules.yaml
Comment (Kaizhe) [inline rules/falco_rules.yaml:3018]: Please add tags fields with `process`

File: rules/falco_rules.yaml
Comment (Kaizhe) [reply rules/falco_rules.yaml:3016]: Agreed with @leogr , need to have a whitelisted images/containers that perform kernel module injection like falco.

**Review by Kaizhe — CHANGES_REQUESTED** [review]

File: rules/falco_rules.yaml
Comment (leodido) [reply rules/falco_rules.yaml:3016]: I'm @leodido anyway 😄

Comment (leogr) [reply rules/falco_rules.yaml:3016]: > I'm @leodido anyway

I believe @Kaizhe referred to [my comment](https://github.com/falcosecurity/falco/pull/1401#pullrequestreview-504589976) where I was suggesting to introduce a `list` :smile_cat: 

Anyway, to summarize, I believe we all agree on two things:
- a list of known images/containers that perform kernel module injection is needed
- the above list must include Falco's images

**Comment by leogr** [comment]
Hey @d1vious 

Could you also rebase this PR, please?

**Comment by d1vious** [comment]
@leogr  rebased as requested. Also added the changes you asked for @Kaizhe

**Comment by leogr** [comment]
> @leogr rebased as requested. Also added the changes you asked for @Kaizhe. Let me know what else I can help with

Hey @d1vious 

Thank you, but unfortunately, I see some issues now. There're merge commits in this PR and commits missing the sign-off that are both not allowed in this repo (please check out the [comments from our bot](https://github.com/falcosecurity/falco/pull/1401#issuecomment-720873229)). Moreover, integration tests are failing. :(

Thus, I'd suggest reverting the merge then use `git rebase -i master` please. If you need any help, feel free to contact me via DM on Slack.

**Comment by d1vious** [comment]
@leogr This literally has been more work than is worth IMHO, if you want to rebase and include the changes feel free. I was hoping to contribute back to the project this detection which I found useful in my production environment at Splunk, but contributing here has been extremely cumbersome due to all the steps/hoops required and will likely not contribute again unfortunately. My advice is make this process easier for contributors if you really want the community to be part of the project. Feel free to close the PR if you did not find it useful, otherwise enjoy the free detection 😄

File: rules/falco_rules.yaml
Comment (Kaizhe) [inline rules/falco_rules.yaml:3061]: `evt.type=execve` is redundant. Can you please remove it?

**Review by Kaizhe — CHANGES_REQUESTED** [review]

**Comment by leogr** [comment]
> @leogr This literally has been more work than is worth IMHO, if you want to rebase and include the changes feel free. I was hoping to contribute back to the project this detection which I found useful in my production environment at Splunk, but contributing here has been extremely cumbersome due to all the steps/hoops required and will likely not contribute again unfortunately. My advice is make this process easier for contributors if you really want the community to be part of the project. Feel free to close the PR if you did not find it useful, otherwise enjoy the free detection

Hey @d1vious 

I'm sorry you felt unhappy with the process we have in place. Your contribution is valuable, and we appreciate it. We never thought it was unuseful. 

However, I'm here to help with this PR. So let's try to solve the issue.
We actually just need to rebase it, then @Kaizhe's suggestions can be applied easily.

Unfortunately, GitHub does not allow us to force push on your branch, so I cannot rebase this PR by myself atm. I can propose two solutions (almost effortless for you):
- giving me collaborator access to your fork so that I can rebase your branch
- or, I can create a new PR if that's ok for you

In both cases, I will cherry-pick and keep your two commits (that have been signed-off by you, so it's perfectly fine) then we will also be able to include our suggestions on top of that.

WDYT?

**Comment by d1vious** [comment]
cleaned up, added the fixes and DCO signature on https://github.com/falcosecurity/falco/pull/1478, closing this one

**Pull request closed without merging**
